docs/Analyzer reference page template.md

```diff
@@ -0,0 +1,31 @@
+# RULEID: Friendly rule name
+
+## Cause
+
+## Rule description
+
+## How to fix violations
+
+## When to suppress warnings
+
+## Example of a violation
```

Comment:
Hmm, seems MSDN uses violation already and we're just following that here, so OK by me. Though we might want to switch to issue to match the SARIF direction. Up to you.